### PR TITLE
#287 - Implementing a Select All for multi select

### PR DESCRIPTION
**Pull request by aaronpowell**
- new property
- If property is true render 'Select All' option
- When clicked take all the filteredOptions and add them
- Added an example
- Updated docs

### CODE REVIEW

File: src/Select.js [hunk 1]
```diff
@@ -659,7 +691,7 @@ var Select = React.createClass({
 
 		var ops = Object.keys(this.state.filteredOptions).map(function(key) {
 			var op = this.state.filteredOptions[key];
-			var isSelected = this.state.value == op.value;
+			var isSelected = this.state.value === op.value;
```
Comment (aaronpowell) [inline src/Select.js]: For the record I _didn't_ need to change this except running `npm run lint` picks it up, and that was already in the `master` branch (https://github.com/JedWatson/react-select/blob/master/src/Select.js#L662) so someone didn't run it before committing :stuck_out_tongue:

**Comment by aaronpowell** [comment]
Is anyone available to have a look at this PR? I'm almost to the point where I'm blocked on how to implement the feature that I'd like to use it with, either using the proposed change or finding an alternate solution.

**Comment by JedWatson** [comment]
@aaronpowell I'm about to jump on a plane home from react conf eu, will get on this when I land.

**Comment by aaronpowell** [comment]
Any movement on this or shall I close it?

**Comment by TienPhan** [comment]
@JedWatson Any chance on an update?

**Comment by FunkyMJ** [comment]
Hey @JedWatson. Can you accept this PR?

**Comment by Handroo** [comment]
Any updates on this feature?

**Comment by involve-it** [comment]
+1 
@JedWatson , did you land from the plane?? :\

**Comment by ashwinurao** [comment]
@JedWatson Any update on this PR? I am also looking for this feature, would love if this was integrated.

**Comment by Carlows** [comment]
+1

**Comment by jfernandogt** [comment]
+1

**Comment by the-dcruz** [comment]
+1

**Comment by chiester** [comment]
+1

**Comment by patrickbeeson** [comment]
👍

**Comment by jorgevenegas** [comment]
+1

**Comment by erlendve** [comment]
+1

**Comment by bahaddinyasar** [comment]
+1

**Comment by vitalytr** [comment]
+1

**Comment by frogzsj** [comment]
+1

**Comment by abhayjeetgupta** [comment]
+1

**Comment by ty525** [comment]
+1

**Comment by tobold** [comment]
Any update on this?

**Comment by aaronpowell** [comment]
I'm closing this, it's years out of date and I have no need for it so have no interest in ensuring it still works.

**Pull request closed without merging**